### ISSUE TITLE
W1b: `RandomMosaic` returns `(W, H)` on a non-square input, and the same axis swap reaches `start_ratio_range`

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4438]
`RandomMosaic` composes a mosaic from a batch and, with the default `output_size=None`, is documented to keep the
input size. It does keep the number of pixels, but it transposes them: a `(B, C, 6, 8)` batch comes back
`(B, C, 8, 6)`. The same swap reaches three further surfaces that all trace to one root cause — the unbatched
`(C, H, W)` path, the `start_ratio_range` slots, and the output shape itself, which changes with the draw because
the swap only happens when the augmentation fires.

`kornia/augmentation/random_generator/_2d/mosaic.py:101-106` hands `input_sizes[0]`, the height, to
`bbox_generator` as its `width` argument and `input_sizes[1]`, the width, as its `height`, so the source crop is
`H` wide and `W` tall; `crop_dst` two lines below uses the correct order. Line 99 of the same file computes
`start_corner_x = start_corner_factor[:, 0] * batch_shape[-2]`, scaling the slot it calls `x` by the height, so
`start_ratio_range`'s first entry moves the box down rather than across. An explicit `output_size` is honoured
correctly as `(H, W)`, which is why the swap is invisible to anyone who passes one.

The class doctest uses a `224x224` input and a `(300, 300)` output size, and every `RandomMosaic` fixture in
`tests/augmentation/test_augmentation_mix.py` and `tests/augmentation/container/test_augmentation_sequential.py`
is square too — `8x8`, `100x100`, `224x224` with `output_size=(300, 300)`. Nothing in the repository looks at a
non-square mosaic, so the swap has never had a chance to fail a check. A caller who feeds one gets a transposed
image and, at `p < 1.0`, a module whose output shape is not a function of its input shape.

## Repro

```python
import torch
import kornia.augmentation as K


for (h, w) in ((6, 8), (8, 6), (5, 7), (4, 4)):
    torch.manual_seed(0)
    out = K.RandomMosaic(p=1.0)(torch.rand(4, 3, h, w))
    print(f"input (H, W) = ({h}, {w})  ->  output {tuple(out.shape)}")

# an explicit output_size IS honoured as (H, W)
torch.manual_seed(0)
print("output_size=(6, 10) on 5x7   ->  output", tuple(K.RandomMosaic(p=1.0, output_size=(6, 10))(torch.rand(4, 3, 5, 7)).shape))

# the shape of one module's output depends on whether the augmentation fires
for p in (0.0, 1.0):
    torch.manual_seed(0)
    print(f"p={p}  ->  output", tuple(K.RandomMosaic(p=p)(torch.rand(4, 3, 6, 8)).shape))

# the unbatched (C, H, W) path carries the same swap
for keepdim in (False, True):
    torch.manual_seed(0)
    print(f"(C, H, W) = (1, 6, 8), keepdim={keepdim}  ->  RandomMosaic  ",
          tuple(K.RandomMosaic(p=1.0, keepdim=keepdim)(torch.rand(1, 6, 8)).shape),
          " RandomMixUpV2 ", tuple(K.RandomMixUpV2(p=1.0, keepdim=keepdim)(torch.rand(1, 6, 8)).shape))

# the source box handed to the crop: 5 wide x 7 tall on a 6x8 input
torch.manual_seed(0)
aug = K.RandomMosaic(p=1.0)
aug(torch.rand(4, 3, 6, 8))
src, dst = aug._params["src"][0], aug._params["dst"][0]
print("src[0] =", [[round(v, 4) for v in pt] for pt in src.tolist()])
print("dst[0] =", [[round(v, 4) for v in pt] for pt in dst.tolist()])

# start_ratio_range: the slot the code calls x is scaled by H
for (h, w) in ((6, 8), (3, 11)):
    for r in (0.0, 0.5, 1.0):
        torch.manual_seed(0)
        a = K.RandomMosaic(p=1.0, start_ratio_range=(r, r))
        a(torch.rand(4, 3, h, w))
        print(f"HxW={h}x{w}  start_ratio_range=({r}, {r})  ->  src[0] top-left =",
              [round(v, 4) for v in a._params["src"][0, 0].tolist()])
```

## Observed

```
input (H, W) = (6, 8)  ->  output (4, 3, 8, 6)
input (H, W) = (8, 6)  ->  output (4, 3, 6, 8)
input (H, W) = (5, 7)  ->  output (4, 3, 7, 5)
input (H, W) = (4, 4)  ->  output (4, 3, 4, 4)
output_size=(6, 10) on 5x7   ->  output (4, 3, 6, 10)
p=0.0  ->  output (4, 3, 6, 8)
p=1.0  ->  output (4, 3, 8, 6)
(C, H, W) = (1, 6, 8), keepdim=False  ->  RandomMosaic   (1, 1, 8, 6)  RandomMixUpV2  (1, 1, 6, 8)
(C, H, W) = (1, 6, 8), keepdim=True  ->  RandomMosaic   (1, 8, 6)  RandomMixUpV2  (1, 6, 8)
src[0] = [[3.4557, 5.2725], [8.4557, 5.2725], [8.4557, 12.2725], [3.4557, 12.2725]]
dst[0] = [[0.0, 0.0], [7.0, 0.0], [7.0, 5.0], [0.0, 5.0]]
HxW=6x8  start_ratio_range=(0.0, 0.0)  ->  src[0] top-left = [0.0, 0.0]
HxW=6x8  start_ratio_range=(0.5, 0.5)  ->  src[0] top-left = [3.0, 4.0]
HxW=6x8  start_ratio_range=(1.0, 1.0)  ->  src[0] top-left = [6.0, 8.0]
HxW=3x11  start_ratio_range=(0.0, 0.0)  ->  src[0] top-left = [0.0, 0.0]
HxW=3x11  start_ratio_range=(0.5, 0.5)  ->  src[0] top-left = [1.5, 5.5]
HxW=3x11  start_ratio_range=(1.0, 1.0)  ->  src[0] top-left = [3.0, 11.0]
```

## Expected

`output_size=None` returns `(B, C, H, W)` for a `(B, C, H, W)` input — the size the docstring already promises —
and `start_ratio_range`'s first slot scales the horizontal start by `W` and the second the vertical start by `H`,
matching their names and matching `bbox_generator`'s `(x_start, y_start, width, height)` signature. The fix is the
argument order at `:101-106` plus the two factors at `:99-100`; the commented-out
`output_size = input_sizes if self.output_size is None else self.output_size` at `:82` is the shape half of the
same line of thought.

This changes the output of every existing non-square caller, including anyone who has compensated for the swap by
passing a transposed `output_size`, so it is a behavior change rather than a silent correction.

## Classification

**Broken output** — a transposed image and a draw-dependent output shape on any non-square input, with no error.
Idiom: 7 (an axis-order slip hidden by symmetric test data) — the class doctest is square, so the defect has never
been visible to an example.

## Disposition

`n/a — single symbol`. All four surfaces are `RandomMosaic` and its generator, fixed by one change to
`kornia/augmentation/random_generator/_2d/mosaic.py`; a focused fix is welcome as a PR. Batch 6 documents the
current `(W, H)` default on the class and pins it, and does not change it.

## Environment

kornia 0.9.0rc1 @ `d6a2c138` (main), torch 2.14.0, python 3.11, cpu. Found during the augmentation conventions
audit (batch 6, tracking issue #4407); the batch documents the behavior above as it is and pins it, and does not
change it.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
